runtime/compiler/wasm/reader_test.go

```diff
@@ -451,7 +451,247 @@ func TestWasmReader_readTypeSection(t *testing.T) {
 	})
 }
 
-func TestWasmReader_readFunctionSection(t *testing.T) {
+func TestWASMReader_readImportSection(t *testing.T) {
+
+	t.Parallel()
+
+	read := func(data []byte) ([]*Import, error) {
+		b := buf{data: data}
+		r := WASMReader{buf: &b}
+		err := r.readImportSection()
+		if err != nil {
+			return nil, err
+		}
+		return r.Module.Imports, nil
+	}
+
+	t.Run("valid", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeIDs, err := read([]byte{
+			// section size: 11 (LEB128)
+			0x8b, 0x80, 0x80, 0x80, 0x0,
+			// import count: 1
+			0x1,
+			// module length
+			0x3,
+			// module = "foo"
+			0x66, 0x6f, 0x6f,
+			// name length
+			0x3,
+			// name = "bar"
+			0x62, 0x61, 0x72,
+			// type indicator: function = 0
+			0x0,
+			// type ID of function: 0
+			0x1,
+		})
+		require.NoError(t, err)
+		assert.Equal(t,
+			[]*Import{
+				{
+					Module: "foo",
+					Name:   "bar",
+					TypeID: 1,
+				},
+			},
+			typeIDs,
+		)
+	})
+
+	t.Run("invalid size", func(t *testing.T) {
+
+		t.Parallel()
+
+		funcTypes, err := read([]byte{
+			0x87, 0x80, 0x80,
+		})
+		require.Error(t, err)
+		assert.Equal(t,
+			InvalidSectionSizeError{
+				Offset:    0,
+				ReadError: io.EOF,
+			},
+			err,
+		)
+		assert.Nil(t, funcTypes)
+	})
+
+	t.Run("invalid count", func(t *testing.T) {
+
+		t.Parallel()
+
+		funcTypes, err := read([]byte{
+			// section size: 7 (LEB128)
```

Comment:
the comment seems off

Reply:
Good catch! This was wrong in several places (due to copy and paste), fixed all cases in db04dad5bee706382ba749a7bb8f1f0f02a13305